### ISSUE TITLE
Invalid account name and room id issue

### ISSUE THREAD

**Issue by fizerkhan** [tristandunn/node-campfire#19]
If i give invalid account name and room id, then it throws following error

```
    var instance = new Campfire({
        token: body.campFireApiToken,
        account: body.campFireAccountName
    });

    instance.join(body.campFireRoomId, function(error, room) {

        if (error || !room) {
            return callback(error || new Error('Invalid credentials.'))
        }

        room.speak(message, callback);
    });
```

```
TypeError: Cannot read property 'id' of undefined
    at new Room (/Users/me/server/node_modules/campfire/lib/campfire/room.js:4:30)
    at null.<anonymous> (/Users/me/server/node_modules/campfire/lib/campfire.js:74:18)
    at IncomingMessage.<anonymous> (/Users/me/server/node_modules/campfire/lib/campfire.js:145:18)
    at IncomingMessage.emit (events.js:117:20)
    at _stream_readable.js:943:16
    at process._tickCallback (node.js:419:13)
```

I could not handle this error. Can you please notify error in callback.

**Comment by tristandunn** [comment]
Released the fix in version `0.2.1`.

**Comment by fizerkhan** [comment]
Thanks for the quick fix.

**Comment by fizerkhan** [comment]
@tristandunn It does not work. you miss the bracket  in `new Error`

```
 if (error || response.room === undefined) {
   return callback(error || new Error("Invalid account or room.");
 }
```

It should be

```
 if (error || (response && !response.room)) {
   return callback(error || new Error("Invalid account or room."));
 }
```

**Comment by tristandunn** [comment]
That's what I get for adding a message to the error after testing it. Pushed a corrected version as `0.2.2`.
